cmpserver/plugin/config.go

```diff
@@ -0,0 +1,92 @@
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/argoproj/argo-cd/v2/common"
+	configUtil "github.com/argoproj/argo-cd/v2/util/config"
+)
+
+const (
+	ConfigManagementPluginKind string = "ConfigManagementPlugin"
+)
+
+type PluginConfig struct {
+	metav1.TypeMeta `yaml:",inline"`
+	Metadata        metav1.ObjectMeta `yaml:"metadata"`
+	Spec            PluginConfigSpec  `yaml:"spec"`
+}
+
+type PluginConfigSpec struct {
+	Version          string  `yaml:"version"`
+	Init             Command `yaml:"init,omitempty"`
+	Generate         Command `yaml:"generate"`
+	Find             Command `yaml:"find"`
+	Check            Command `yaml:"check"`
```

Comment:
@alexmt, I do not see this field `Check` used. Want to confirm with you that I can delete it.